### PR TITLE
Makefile to build Canasta-CLI

### PR DESCRIPTION
**Pull request by gitmapd**
Makefile:
- Builds the canasta-cli go project.
- Checks syntax errors.

### CODE REVIEW

**Comment by jeffw16** [comment]
I want to get a review done by @hexmode first if possible, since he is more experienced with Makefiles.

**Comment by hexmode** [comment]
> I want to get a review done by @hexmode first if possible, since he is more experienced with Makefiles.

I'll try to look at this later this week.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,40 @@
+APP_NAME="canasta"
+
+build:
```
Comment (bawolff) [inline Makefile:3]: Should these targets be declared as .PHONY ?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+build: 
+        @go build -o ${APP_NAME} ./.
+        @go mod tidy
+        @go mod verify
+        @echo "To run the CLI you have just generated, please call it as \033[0;32m./${APP_NAME}\033[0m instead of ${APP_NAME}."
+
+test: 
+        go test -race -cover ./...
+
+benchmark:
```
Comment (bawolff) [inline Makefile:12]: Kind of odd to have a benchmark target when there are no benchmarks.

**Review by bawolff — CHANGES_REQUESTED** [review]
You need to use tabs not spaces. This is not just a stylistic difference in makefiles.

This seems to have a lot of targets that don't do anything (benchmark, test, etc). I suppose they might in future though. I'm not sure if that is wanted.

I think this should  include the extra ldflags like build.sh does.

The makefile seems to unconditionally rebuild, even if no files have changed. I suppose that is intentional?

**Comment by hexmode** [comment]
@bawolff thanks for reviewing this. I lost track of it.

**Comment by yaronkoren** [comment]
An improved version of this patch was implemented in #100.

**Pull request closed without merging**